### DISCUSSION TITLE
Improved requests timings helper class

### DISCUSSION THREAD

**Discussion by davenquinn** [developmentseed/titiler#424] [Feature Request]
When using TiTiler within https://github.com/davenquinn/mars-tiler, I reimplemented several aspects of the `MosaicTilerFactory` to play nicely with PostgreSQL database querying for image footprints. I am currently trying to re-align some aspects of this work with the core module, and am factoring out customizations.

One aspect of TiTiler's core that I have heavily modified is the `Timer` class that is used for generating debug timings. This class currently works in the context of a single rendering function, which limits its usefulness for reporting the timings of processing within renderers or post-processing subroutines.

Mars-tiler's [`Timer` class replacement](https://github.com/davenquinn/mars-tiler/blob/main/mars_tiler/timer.py) uses Context variables to allow timings to be easily tracked in subroutines if a timer is available. It also centralizes some of the time-delta math and boilerplate for creating the appropriate response headers.

Would you accept a pull request to integrate this functionality into TiTiler core?

**Comment by vincentsarago** [comment]
👋 @davenquinn, sure why not 🤷‍♂️
